uitools/import/Esri/ArcGISRuntime/Toolkit/Controller/SearchViewController.qml

```diff
@@ -14,6 +14,509 @@
+                    }
+                }
+            }
+        }
+
+        property ListModel suggestions: ListModel {
+            function element(index) {
+                return this.get(index.row);
+            }
+        }
+
+        property ListModel results: ListModel {
+            function element(index) {
+                return this.get(index.row);
+            }
+        }
+
+        property Connections geoViewConnections: Connections {
+            target: controller.geoView
+            function onViewpointChanged() {
+                if (!controller.automaticConfigurationEnabled) {
+                    return;
+                }
+
+                // update queryArea/queryCenter with the viewpoint change.
+                if (geoView instanceof SceneView) {
+                    const vp = geoView.currentViewpointCenter;
+                    if (vp) {
+                        internal.queryCenter = vp.center;
+                        internal.queryArea = null;
+                    }
+                } else if (geoView instanceof MapView) {
+                    const vp2 = geoView.currentViewpointExtent;
+                    if (vp2) {
+                        internal.queryCenter = vp2.extent.center;
+                        internal.queryArea = vp2.extent;
+                    }
+                }
+
+                // Show the requery button if the user manually navigated the view.
+                if (geoView.navigating) {
+                    controller.eligableForRequery = true;
+                }
+            }
+            function onNavigatingChanged() {
+                console.log("Navigating", geoView.navigating);
+            }
+        }
+
+        property Component defaultWorldGeocoderSource: Component {
+            LocatorSearchSource {
+                locatorTask: LocatorTask {
+                    url: 'https://geocode-api.arcgis.com/arcgis/rest/services/World/GeocodeServer'
+                }
+            }
+        }
+
+        function onSuggestionsAdded(source, first, last) {
+            for (let i = first; i <= last; i++) {
+                const suggestion = source.get(i);
+                controller.suggestions.append(suggestion);
+            }
+        }
+
+        function onSuggestionsRemoved(source, first, last) {
+            if (controller.suggestions.count > 0) {
+                for (let i = controller.suggestions.count - 1; i >= 0; i--) {
+                    for (let j = last;j >= first; j--) {
+                        const s1 = source.get(j);
+                        const s2 = controller.suggestions.get(i);
+                        // Test suggestion equallity. If suggestion matches then we remove it.
+                        if (s1.displayTitle === s2.displayTitle &&
+                                s1.displaySubtitle === s2.displaySubtitle &&
+                                s1.collection === s2.collection &&
+                                s1.owningSource === s2.owningSource) {
```

Comment:
Can you implement your own equals for this type?

Reply:
At the API level you mean?

Reply:
No, I meant in the toolkit code. This type looks like a Toolkit object. If it is possible to add an equivalence operator to the API to support this, consider logging an issue for it.